In [2]:
import spacy

In [3]:
nlp = spacy.load('en_core_web_md')

In [4]:
word1 = nlp("Banana")
word2 = nlp("Cat")
word3 = nlp("Monkey")

print(word1.similarity(word2))
print(word2.similarity(word3))
print(word1.similarity(word3))

0.3325463550139417
0.6340912166013265
0.4777847588161717


In [9]:
#tokens = nlp('cat apple monkey banana')
tokens = nlp ('cat milk monkey banana')
for token1 in tokens:
    for token2 in tokens:
        print(token1.text,token2.text,token1.similarity(token2))

cat cat 1.0
cat milk 0.22972984611988068
cat monkey 0.5929929614067078
cat banana 0.2235882729291916
milk cat 0.22972984611988068
milk milk 1.0
milk monkey 0.16944406926631927
milk banana 0.48031964898109436
monkey cat 0.5929929614067078
monkey milk 0.16944406926631927
monkey monkey 1.0
monkey banana 0.404150128364563
banana cat 0.2235882729291916
banana milk 0.48031964898109436
banana monkey 0.404150128364563
banana banana 1.0


As decribed in the pdf the same observed from the monkey and banana(monkey banana 0.404150128364563) though they are in different group fruit and animal. it create a curiosity to find the similarity between cat and milk(cat milk 0.22972984611988068) but it disappointed me.

In [8]:
sentence_to_compare = "Why my cat is on the car"
sentences =["Wher did my dog go",
           "Hello, There is my car",
           "I\'ve lost my car in my car",
           "I \'d like my boat back",
            "I will name my dog Diana"]
model_sentence = nlp(sentence_to_compare)

for sentence in sentences:
    similarity = nlp(sentence).similarity(model_sentence)
    print(sentence + ' -',similarity)

Wher did my dog go - 0.6085941257210434
Hello, There is my car - 0.7914211168930609
I've lost my car in my car - 0.6787541521638475
I 'd like my boat back - 0.5624940476146296
I will name my dog Diana - 0.6491444691953406
